spec/models/course_year_spec.rb

```diff
@@ -27,21 +27,52 @@
   end
 
   describe "course_modules" do
+    before :each do
+      @teacher = FactoryBot.create(:user, :early_career_teacher)
+
+      @course_year = FactoryBot.create(:course_year)
+      @course_module_one = FactoryBot.create(:course_module, title: "One", course_year: @course_year)
+      @course_lesson_one = FactoryBot.create(:course_lesson, course_module: @course_module_one)
+      @course_lesson_two = FactoryBot.create(:course_lesson, course_module: @course_module_one)
+
+      @course_lesson_one_progress = FactoryBot.create(
+        :course_lesson_progress, course_lesson: @course_lesson_one, early_career_teacher_profile: @teacher.early_career_teacher_profile
+      )
+      @course_lesson_two_progress = FactoryBot.create(
+        :course_lesson_progress, course_lesson: @course_lesson_two, early_career_teacher_profile: @teacher.early_career_teacher_profile
+      )
+    end
+
     it "returns modules in order after they get updated" do
-      course_year = FactoryBot.create(:course_year)
-      course_module_one = FactoryBot.create(:course_module, title: "One", course_year: course_year)
-      course_module_two = FactoryBot.create(:course_module, title: "Two", course_year: course_year)
-      course_module_three = FactoryBot.create(:course_module, title: "Three", course_year: course_year)
-      course_module_four = FactoryBot.create(:course_module, title: "Four", course_year: course_year)
+      course_module_two = FactoryBot.create(:course_module, title: "Two", course_year: @course_year)
+      course_module_three = FactoryBot.create(:course_module, title: "Three", course_year: @course_year)
+      course_module_four = FactoryBot.create(:course_module, title: "Four", course_year: @course_year)
 
-      course_module_two.update!(previous_module: course_module_one)
+      course_module_two.update!(previous_module: @course_module_one)
       course_module_four.update!(previous_module: course_module_two)
       course_module_three.update!(previous_module: course_module_four)
 
-      expected_lessons_with_order = [course_module_one, course_module_two, course_module_four, course_module_three]
-      course_year.course_modules_in_order.zip(expected_lessons_with_order).each do |actual, expected|
+      expected_lessons_with_order = [@course_module_one, course_module_two, course_module_four, course_module_three]
+      @course_year.course_modules_in_order.zip(expected_lessons_with_order).each do |actual, expected|
         expect(actual).to eq(expected)
       end
     end
+    it "returns a progress of not_started when all lessons in that module are not_started" do
+      module_progress = @course_year.modules_with_progress(@teacher).first
+      expect(module_progress.progress).to eql("not_started")
+    end
+    it "returns a progress of in_progress when one lesson in that module is in_progress" do
+      @course_lesson_one_progress.update!(progress: "in_progress")
```

Comment:
`@ course_lesson_one_progress.in_progress!` will do the same. As Alfred put it, power of enum :O